core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -255,6 +255,11 @@ public boolean isNameEditable() {
     @RequirePOST
     @Restricted(NoExternalUse.class)
     public HttpResponse doConfirmRename(@QueryParameter String newName) throws IOException {
+
+        if (!isNameEditable()) {
+            throw new IllegalArgumentException("Trying to rename an item that has not editable name.");
```

Comment:
This looks very user unfriendly. Does the UI really allow users to get this far?

Reply:
I am not sure that the UI permit users to do that, I just wanted to stop processing as soon as possible when it occurs. Anyway the same IllegalArgumentException will be raised just after on https://github.com/jenkinsci/jenkins/blob/217c02bece7afab237d8df669a6a21c80ac740a6/core/src/main/java/hudson/model/AbstractItem.java#L264

Maybe it's more user friendly to do something on https://github.com/jenkinsci/jenkins/blob/217c02bece7afab237d8df669a6a21c80ac740a6/core/src/main/java/hudson/model/AbstractItem.java#L277 and then return a FormValidationError.
@daniel-beck do you think it's better to remove this check or return a FormValidationError ?

Reply:
I changed with a form validation error.

Reply:
> This looks very user unfriendly. Does the UI really allow users to get this far?

generally not, you will not see the link for a rename action.  It was just a belt and braes approach